### PR TITLE
Bump Apache Spark to 2.4.4

### PR DESCRIPTION
**Pull request by Fokko**
The latest and greatest

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ Iceberg's Spark integration is compatible with the following Spark versions:
 | --------------- | ------------- |
 | 0.2.0+ *        | 2.3.0         |
 | 0.3.0+ *        | 2.3.2         |
-| master branch   | 2.4.0         |
+| master branch   | 2.4.4         |
```
Comment (xabriel) [inline README.md]: I think we should not drop support of 2.4.0.

**Comment by Fokko** [comment]
The CI is failing, but it passed locally. I'll look into that.

Just to get the latest. Updating from 2.4.4 to 2.4.0 should not contain any breaking changes to the API, so from an Iceberg perspective, we should be safe.

File: README.md (hunk 1 above)
Comment (Fokko) [reply README.md]: Maybe change it to `2.4.x`?

**Comment by mykolasmith** [comment]
Support for Scala 2.11 is deprecated as of Spark 2.4.4.
Since there will be no Scala 2.11 support in Spark 3.0, I think it makes sense to drop support for versions earlier than Spark 2.4.3. This will pressure users to upgrade to Scala 2.12, which will make it less painful to upgrade users to Spark 3.0 once it comes out.

**Comment by jerryshao** [comment]
>Since there will be no Scala 2.11 support in Spark 3.0, I think it makes sense to drop support for versions earlier than Spark 2.4.1. This will pressure users to upgrade to Scala 2.12, which will make it less painful to upgrade users to Spark 3.0 once it comes out.

We internally use 2.4.x but we're still stick on Scala 2.11, I don't think we have to drop the support of Scala 2.11 for Spark 2.4.x, Iceberg Spark DS could ship with two versions of Scala, as what other projects do.

**Comment by Fokko** [comment]
I believe we shouldn't drop Scala 2.11. It is still widely used, and Spark 3.0 is quite far away. When releasing Iceberg I think we should cross-build the Scala versions. If different versions of Spark are really important to use, we could also extend the Travis build matrix to make sure that is tested against multiple version of Spark.

**Comment by rdblue** [comment]
Iceberg should support all 2.4.x versions, which includes some that use Scala 2.11. In the long term, I don't think that Iceberg should be a factor in deciding which version of Scala to use with Spark. Right now we have Scala 2.11 code. Maybe we should convert that to Java to avoid possible problems for downstream users.

**Comment by rdblue** [comment]
I'm going to merge this.

Originally, I wasn't sure whether to update to 2.4.4 or to continue using 2.4.0 because we want to be compatible with all 2.4.x releases. Continuing to use 2.4.0 seems safer because 2.4.4 should be compatible with it and using the older release prevents relying on new methods that were added since 2.4.0. But, we also use non-public APIs and we want to make sure that those don't break. So I think the safest option is to rely on the latest release because the older ones won't change.

**Comment by rdblue** [comment]
Thanks for working on this, @Fokko!

**Pull request merged**